### ISSUE TITLE
How was the conversion done?

### ISSUE THREAD

**Issue by Beliavsky** [urbanjost/M_blas#1]
Question, not an issue -- did you use a program to convert the BLAS code to free source form with argument intents? If so, could you release the program, since there are many other FORTRAN 77 programs that need conversion?

**Comment by urbanjost** [comment]
No, I made a bunch of things on the fly using vim(1) and sed(1) and grep(1) and bash(1) and did a lot by eye. I did make a little program to convert to free-format but pretty sure (I know there used to be) there are already converters lthat do more than mine did like changing all loops to end in ENDDO.  I believe that LFORTRAN is planning to do that. I used the compilers to help verify some of the intents and did make a VERY kludgy script to help identify the intents.  I thought the option to automatically generate interface files added INTENT but it did not, at least with compilers I tried.  It would be really useful, but I don't have any; and considering the lists you have put together and updated, especially on the Fortran WIki, I don't think there is anything, as that requires not just simple parsing of the code but of any procedure calls that are passed parameters as well. Since LFORTRAN requires parsing all the code a feature like that would be feasible if it isn't planned.
